src/contracts/GPv2Settlement.sol

```diff
@@ -44,6 +44,15 @@ contract GPv2Settlement {
     /// contract is created during deployment
     GPv2AllowanceManager internal immutable allowanceManager;
 
+    /// @dev Map each user order to the amount that has been filled so far. If
+    /// this amount is larger than or equal to the amount traded in the order
+    /// (amount sold for sell orders, amount bought for buy orders) then the
+    /// order cannot be traded anymore. If the order is fill or kill, then this
+    /// value is only used to determine whether the order has already been
+    /// executed.
+    /// See [`orderId`] for how to represent an order with a single bytes32.
+    mapping(bytes32 => uint256) public filledAmount;
```

Comment:
I would make this `internal` for now, we can always make it public later, but I suspect we won't need to as we can without much extra work compute the storage slot for the filled amount.

Reply:
Is this assuming we use StorageAccessible as a base class? Otherwise how would other smart contracts be able to access this information. What's the cost of leaving it public?

Reply:
> Is this assuming we use StorageAccessible as a base class?

Yes that was my assumption.

Reply:
Any reason to choose `internal` instead of `private`? I should be able to read storage slots from JS.

Reply:
> Any reason to choose internal instead of private

My main argument was that `interal` allows it to be used from the `TestInterface` contract. IMO it makes the test a bit clearer compare to using `eth_getStorageAt`.

Reply:
I'd use a wrapper function to read from storage in case. I think we should write the contract "as it should be" and not "as it's nice to test" in principle, `internal` is kind of useless for us here.
Just out of curiosity, I tested the gas consumption of a call to `invalidateOrder` in the case `filledAmount` is:
1. private: 42145
2. internal: 42145 (same as private)
3. public: 42100 (yes, it's lower!)

Definitely not saying that we should use `public` because of this, but it's quite a surprising finding.

Reply:
> I think we should write the contract "as it should be" and not "as it's nice to test" in principle, internal is kind of useless for us here.

For me, `internal` is like `private` except inherited contracts can also call those functions, which is nice IMO.

> public: 42100 (yes, it's lower!)

That's incredibly surprising! I can't wait for ecosystem Friday :tada:

Reply:
> For me, internal is like private except inherited contracts can also call those functions, which is nice IMO.

Mmh. I see, my mental distinction between `private` and `internal` was quite arbitrary. This makes sense.
I agree with `internal` as well as almost all choices of `internal`/`private` in this contract (except `authenticator`, which should be internal). I'll apply these changes.

Reply:
Pardon my ignorance, but why do we want to make it private/internal? One learning form the v1 contracts (where e.g. balances was made private) is that it just makes accessing this value harder later on (even if we use tools like StorageAccessible we need to deploy extensions or compute storage slots locally which is not effort-less).

Reply:
Sounds good to me.

Reply:
My belief is that making functions internal makes them more gas efficient, because
1. a good compiler could be inlining(?) the functions so that, for example, the order of the objects in memory in the caller function is already optimized to be used in the called function, instead of rearranging things.
2. each new extra public method makes calling the old functions slightly more expensive.
Both are debatable: the first point is contradicted by my gas benchmark from before, and the only function that we need to really make gas efficient is `settle`, which we can make callable from the fallback function so that it's not affected by adding new functions.

For this PR, I'd keep everything internal/private for consistency with the rest anyway, and leave it to a separate PR to make everything public.

Reply:
I'm OK with that too. Then we can introduce the 56-byte orderUuid as well.

Reply:
I agree with this argument for functions, but here we are talking about state. Why is it more gas efficient to have the mapping declared as private?

Reply:
I was under the impression @fedgiac was talking about functions and 
```solidity
mapping(bytes32 => uint256) public filledAmount;
```

Should stay as is, just because its always better to enable accessing contract storage on the EVM for future proofing (lessons learnt from GPv1 and balances).

Reply:
Ok, in that case I misunderstood. Judging from the location of the discussion and the initial comment in this thread I was under the impression we are discussing the visibility of this mapping. I'm happy if we keep the mapping public.

Reply:
I might have mixed the threads as they got quite long :see_no_evil:. I think your interpretation was very much justified given the location of the comment. Glad my comment is cleared up now :smile: